solidity-checks/src/main/java/org/sonarsource/solidity/checks/CurlyBraceCheck.java

```diff
@@ -0,0 +1,148 @@
+    BlockContext functionBlock = ctx.block();
+    int functionLastLine = ctx.modifierList().getStop().getLine();
+    if (ctx.returnParameters() != null) {
+      functionLastLine = ctx.returnParameters().getStop().getLine();
+    }
+    if (functionBlock != null) {
+      Token curlyBraceToken = (Token) functionBlock.getChild(0).getPayload();
+      if (curlyBraceToken.getLine() != functionLastLine) {
+        report(curlyBraceToken);
+      }
+    }
+    return super.visitFunctionDefinition(ctx);
+  }
+
+  @Override
+  public ParseTree visitForStatement(ForStatementContext ctx) {
+    StatementContext statementList = ctx.statement();
+    if (CheckUtils.treeMatches(statementList.getChild(0), BlockContext.class)) {
+      List<ExpressionContext> expressionList = ctx.expression();
+      Token curlyBraceToken = (Token) statementList.block().getChild(0).getPayload();
+      if (curlyBraceToken.getLine() != expressionList.get(expressionList.size() - 1).getStop().getLine()) {
+        report(curlyBraceToken);
+      }
+    }
+    return super.visitForStatement(ctx);
+  }
+
+  @Override
+  public ParseTree visitDoWhileStatement(DoWhileStatementContext ctx) {
+    StatementContext statementList = ctx.statement();
+    if (CheckUtils.treeMatches(statementList.getChild(0), BlockContext.class)) {
+      TerminalNode doKeyword = ctx.doKeyWord();
+      Token curlyBraceToken = (Token) statementList.block().getChild(0).getPayload();
+      if (doKeyword.getSymbol().getLine() != curlyBraceToken.getLine()) {
+        report(curlyBraceToken);
+      }
+    }
+    return super.visitDoWhileStatement(ctx);
+  }
+
+  @Override
+  public ParseTree visitWhileStatement(WhileStatementContext ctx) {
+    StatementContext statementList = ctx.statement();
+    if (CheckUtils.treeMatches(statementList.getChild(0), BlockContext.class)) {
+      ExpressionContext expression = ctx.expression();
+      Token curlyBraceToken = (Token) statementList.block().getChild(0).getPayload();
+      if (expression.getStop().getLine() != curlyBraceToken.getLine()) {
+        report(curlyBraceToken);
+      }
+    }
+    return super.visitWhileStatement(ctx);
+  }
+
+  @Override
+  public ParseTree visitStructDefinition(StructDefinitionContext ctx) {
+    Token curlyBraceToken = ctx.getOpenCurlyBrace().getSymbol();
+    if (ctx.identifier().getStart().getLine() != curlyBraceToken.getLine()) {
+      report(curlyBraceToken);
+    }
+    return super.visitStructDefinition(ctx);
+  }
+
+  @Override
+  public ParseTree visitModifierDefinition(ModifierDefinitionContext ctx) {
+    Token curlyBraceToken = (Token) ctx.block().getChild(0).getPayload();
+    int modifierLastLine = ctx.identifier().Identifier().getSymbol().getLine();
+    if (ctx.parameterList() != null && ctx.parameterList().getStop().getLine() != curlyBraceToken.getLine()) {
+      report(curlyBraceToken);
+    } else if (modifierLastLine != curlyBraceToken.getLine()) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This branch's code block is the same as the block for the branch on line 127. ([squid:S1871](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1871))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=36&issues=AWR_VuU9l6vpliCeI1ps&open=AWR_VuU9l6vpliCeI1ps)

[ISSUE_KEY]: AWR_VuU9l6vpliCeI1ps